transformer_engine/jax/csrc/modules.cpp

```diff
@@ -211,107 +235,93 @@ void GeluFP8(cudaStream_t stream, void **buffers, const char *opaque, size_t opa
 }
 
-void DGelu(cudaStream_t stream, void **buffers, const char *opaque, size_t opaque_len) {
+void DActLu(cudaStream_t stream, void **buffers, const char *opaque, size_t opaque_len) {
     auto *input = buffers[0];
-    auto *gelu_input = buffers[1];
+    auto *act_input = buffers[1];
     auto *output = buffers[2];
 
     const auto &desc = *UnpackOpaque<CustomCallCommonDescriptor>(opaque, opaque_len);
     auto m = desc.shape.dims[0];
     auto n = desc.shape.dims[1];
+    auto act_enum = desc.act_enum;
+
+    // Gated should have odd enum value
+    auto act_len = act_enum % 2 + 1;
     auto input_shape = std::vector<size_t>{m, n};
-    auto gelu_input_shape = std::vector<size_t>{m, n};
-    auto output_shape = std::vector<size_t>{m, n};
+    auto act_input_shape = std::vector<size_t>{m, n * act_len};
+    auto output_shape = std::vector<size_t>{m, n * act_len};
 
     auto input_tensor = TensorWrapper(input, input_shape, desc.in_dtype);
-    auto gelu_input_tensor = TensorWrapper(gelu_input, gelu_input_shape, desc.in_dtype);
+    auto act_input_tensor = TensorWrapper(act_input, act_input_shape, desc.in_dtype);
     auto output_tensor = TensorWrapper(output, output_shape, desc.out_dtype);
 
-    nvte_dgelu(input_tensor.data(), gelu_input_tensor.data(), output_tensor.data(), stream);
-}
-
-void DGeluDBiasCastTranspose(cudaStream_t stream, void **buffers, const char *opaque,
-                             size_t opaque_len) {
-    auto *input = buffers[0];
-    auto *gelu_input = buffers[1];
-    float *amax = reinterpret_cast<float *>(buffers[2]);
-    float *scale = reinterpret_cast<float *>(buffers[3]);
-    float *scale_inv = reinterpret_cast<float *>(buffers[4]);
-    auto *output = buffers[5];
-    auto *output_trans = buffers[6];
-    auto *dbias = buffers[7];
-    float *amax_out = reinterpret_cast<float *>(buffers[8]);
-    void *workspace_ptr = buffers[9];
-
-    const auto &desc = *UnpackOpaque<CustomCallCommonWkDescriptor>(opaque, opaque_len);
-    assert(amax == amax_out);
-    if (!use_fp8(desc.out_dtype)) {
-        scale = nullptr;
-        scale_inv = nullptr;
-        amax_out = nullptr;
+    switch (act_enum) {
+      case ActivationEnum::GELU:
+        nvte_dgelu(input_tensor.data(), act_input_tensor.data(),
+                   output_tensor.data(), stream);
+        break;
+      case ActivationEnum::GEGLU:
+        nvte_dgeglu(input_tensor.data(), act_input_tensor.data(),
+                    output_tensor.data(), stream);
+        break;
+      case ActivationEnum::SILU:
+        nvte_dswish(input_tensor.data(), act_input_tensor.data(),
+                    output_tensor.data(), stream);
+        break;
+      case ActivationEnum::SWIGLU:
+        nvte_dswiglu(input_tensor.data(), act_input_tensor.data(),
+                     output_tensor.data(), stream);
+        break;
+      default:
+        // TODO: what is a good message here
+        throw std::runtime_error("Not Implemented");
```

Comment:
There is `NVTE_ERROR` marco for raising error for unexpected behavior.
For example, you can use it here as
```
NVTE_ERROR("Unsupported ActivationEnum");
```